pygame/math.py

```diff
@@ -0,0 +1,1529 @@
+        return NotImplemented
+
+    def __rsub__(self, other):
+        if isinstance(other, Number):
+            return Vector2(other - self.vector.x, other - self.vector.y)
+        elif isinstance(other, Vector2):
+            return Vector2(other.x - self.vector.x, other.y - self.vector.y)
+        return NotImplemented
+
+    def __mul__(self, other):
+        if isinstance(other, Number):
+            return Vector2(self.vector.x * other, self.vector.y * other)
+        elif isinstance(other, Vector2):
+            return Vector2(self.vector.x * other.x, self.vector.y * other.y)
+        elif isinstance(other, ElementwiseVector2Proxy):
+            return Vector2(self.vector.x * other.vector.x,
+                           self.vector.y * other.vector.y)
+        return NotImplemented
+
+    def __truediv__(self, other):
+        if isinstance(other, Number):
+            return Vector2(self.vector.x / other, self.vector.y / other)
+        elif isinstance(other, Vector2):
+            return Vector2(self.vector.x / other.x, self.vector.y / other.y)
+        elif isinstance(other, ElementwiseVector2Proxy):
+            return Vector2(self.vector.x / other.vector.x,
+                           self.vector.y / other.vector.y)
+        return NotImplemented
+
+    __div__ = __truediv__  # for backwards compatibility with Python 2
+
+    def __rtruediv__(self, other):
+        if isinstance(other, Number):
+            return Vector2(other / self.vector.x, other / self.vector.y)
+        elif isinstance(other, Vector2):
+            return Vector2(other.x / self.vector.x, other.y / self.vector.y)
+        return NotImplemented
+
+    __rdiv__ = __rtruediv__  # for backwards compatibility with Python 2
+
+    def __floordiv__(self, other):
+        if isinstance(other, Number):
+            return Vector2(self.vector.x // other, self.vector.y // other)
+        elif isinstance(other, Vector2):
+            return Vector2(self.vector.x // other.x, self.vector.y // other.y)
+        elif isinstance(other, ElementwiseVector2Proxy):
+            return Vector2(self.vector.x // other.vector.x,
+                           self.vector.y // other.vector.y)
+        return NotImplemented
+
+    def __rfloordiv__(self, other):
+        if isinstance(other, Number):
+            return Vector2(other // self.vector.x, other // self.vector.y)
+        elif isinstance(other, Vector2):
+            return Vector2(other.x // self.vector.x, other.y // self.vector.y)
+        return NotImplemented
+
+    def __pow__(self, other):
+        if isinstance(other, Number):
+            return Vector2(self.vector.x ** other, self.vector.y ** other)
+        elif isinstance(other, Vector2):
+            return Vector2(self.vector.x ** other.x, self.vector.y ** other.y)
+        elif isinstance(other, ElementwiseVector2Proxy):
+            return Vector2(self.vector.x ** other.vector.x,
+                           self.vector.y ** other.vector.y)
```

Comment:
Thanks @drnlm.  I have done something like that in f849efd.  Please have a look.

I guess I should add a copyright block as well for math.py.  Whose names do I include?
For math_test.py, most of it came from the original pygame repo, so leave it as is?